src/test/system/borrower/repay/creditline_repay.t.sol

```diff
@@ -78,15 +93,40 @@ contract PrincipalRepayTest is BaseSystemTest {
         //repay after 1 year
         hevm.warp(block.timestamp + 365 days);
         assertPreCondition(loanId, tokenId, repayAmount, expectedDebt);
-      //  repay(loanId, tokenId, repayAmount, expectedDebt);
+        repay(loanId, tokenId, repayAmount, expectedDebt);
     }
 
+    function testCreditLineCycle() public {
+        uint nftPrice = 200 ether; // -> ceiling 100 ether
+        uint riskGroup = 1; // -> 12% per year
+        uint maxCeiling = 100 ether;
+
+        // expected debt after 1 year of compounding
+        (uint loanId, uint tokenId) = createLoanAndWithdraw(borrower_, nftPrice, riskGroup); // borrow max amount
+      
+        // supply borrower with additional funds to pay for accrued interest
+        topUp(borrower_);
+        // borrower allows shelf full control over borrower tokens
+        Borrower(borrower_).doApproveCurrency(address(shelf), type(uint256).max);
+        //repay after 1 year
+        assertEq(nftFeed.ceiling(loanId), 0); // ceiling = 0 after borrow
+        hevm.warp(block.timestamp + 365 days);
+        assertEq(nftFeed.ceiling(loanId), 0); // ceiling = 0 after interest accrued
+        uint repaymentAmount = 50 ether;
+        
+        repay(loanId, tokenId, repaymentAmount, pile.debt(loanId)); // repay half debt
+        assertEq(nftFeed.ceiling(loanId), 37999999999999999999); // ceiling 100 - 62 ether
+
+        repay(loanId, tokenId,  pile.debt(loanId), pile.debt(loanId)); // repay rest of debt
+        assertEq(nftFeed.ceiling(loanId), 100 ether);
+    } 
+
     function testRepayFullDebt() public {
         uint nftPrice = 200 ether; // -> ceiling 100 ether
         uint riskGroup = 1; // -> 12% per year
 
         // expected debt after 1 year of compounding
-        uint expectedDebt = 112 ether;
+        uint expectedDebt = 112000000000000000001; // 112 ETH
```

Comment:
```suggestion
        uint expectedDebt = (112 ether) + 1;
```

Reply:
Why are we here 1 Wei off?

Reply:
our usual rounding errors, that's why the pre/post condition assertions have the 1 wei tolerance